frontend/public/components/cluster-settings/openid-idp-form.tsx

```diff
@@ -0,0 +1,272 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+
+import { OAuthModel, SecretModel } from '../../models';
+import { k8sCreate, k8sGet, k8sPatch, K8sResourceKind, referenceFor } from '../../module/k8s';
+import {
+  AsyncComponent,
+  ButtonBar,
+  Dropdown,
+  PromiseComponent,
+  history,
+  resourceObjPath,
+} from '../utils';
+
+// The name of the cluster-scoped OAuth configuration resource.
+const oauthResourceName = 'cluster';
+
+const DroppableFileInput = (props) => <AsyncComponent loader={() => import('../utils/file-input').then(c => c.DroppableFileInput)} {...props} />;
+
+export class AddOpenIDPage extends PromiseComponent {
+  readonly state: AddOpenIDPPageState = {
+    name: 'openidconnect',
+    mappingMethod: 'claim',
+    clientID: '',
+    clientSecretFileContent: '',
+    preferredUserName: 'preferred_username',
+    claimName: 'name',
+    claimEmail: 'email',
+    issuer: '',
+    caFileContent: '',
+    extraScopes: '',
+    extraAuthorizeParameters: '',
+    inProgress: false,
+    errorMessage: '',
+  };
+
+  getOAuthResource(): Promise<K8sResourceKind> {
+    return this.handlePromise(k8sGet(OAuthModel, oauthResourceName));
+  }
+
+  createOIDClientSecret(): Promise<K8sResourceKind> {
+    const secret = {
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        generateName: 'openid-',
```

Comment:
I'd give this a more specific name.

```suggestion
        generateName: 'openid-client-secret-',
```